include/swift/AST/DistributedDecl.h

```diff
@@ -31,6 +31,17 @@ class DeclContext;
 class FuncDecl;
 class NominalTypeDecl;
 
+/// Determine if this distributed actor can synthesize a `Codable` conformance.
+/// This is based on the actor's `ID` being `Codable`.
+///
+///  It is possible for the `ID` to be `Codable` but the
+/// `SerializationRequirement` used by the actor (and its actor system to not
+/// be `Codable`). In such situation the conformance is synthesized, however
+/// the user may need to provide an explicit conformance to the
+/// `SerializationRequirement` if they wanted to pass the actor to distributed
+/// methods.
```

Comment:
My thinking here was to drive this off the ID because "okey, we CAN synthesize", so let's.
You're right about the primary reason being that we want this to pass around the actor... 

So in generic code we'd still need to do `DA: DistributedActor & Codable`...

I'm a bit nervous about sneakily introducing "if the `SerReq` is Codable, the ID MUST be" still... I know we talked it through and we could probably get away with it because today we ALWAYS added the conformance after all 🤔 

Maybe I should do a follow up exploration and PR trying to allow this:

```
protocol DA: DistributedActor where ActorSystem: DistributedActorSystem<any Codable> {}

let da: any DA = 
let c: any Codable = da
```

because that's what we'd want to enable I suppose? 🤔 


--- 

The problem with Codable perhaps worth reminding here is that users CANNOT implement the conformance manually: because they'd need to `self = Self.resolve(decoded id)` in the `init(from:)` that `Codable` requires. We special case to allow the self assignment but in general this is not legal in Swift:

```

      // FIXME(distributed): pending swift-evolution, allow `self =` in class
      //  inits in general.
      //  See also: https://github.com/apple/swift/pull/19151 general impl
      auto ext = dyn_cast<ExtensionDecl>(AFD->getDeclContext());
      auto distProto =
          Ctx.getProtocol(KnownProtocolKind::DistributedActor);
      if (distProto && ext && ext->getExtendedNominal() &&
          ext->getExtendedNominal()->getInterfaceType()
              ->isEqual(distProto->getInterfaceType())) {
        auto name = CD->getName();
        auto params = name.getArgumentNames();
        if (params.size() == 1 && params[0] == Ctx.Id_from) {
          // FIXME(distributed): this is a workaround to allow init(from:) to
          //  be implemented in AST by allowing the self to be mutable in the
          //  decoding initializer. This should become a general Swift
          //  feature, allowing this in all classes:
          //  https://forums.swift.org/t/allow-self-x-in-class-convenience-initializers/15924
          selfAccess = SelfAccessKind::Mutating;
        }
      }
```

Reply:
I should have put more emphasis on the last sentence. I think it’s fine to synthesize it but if the `SerializationRequirement` is not Codable does anybody care that actor is Codable, as in ActorID could be transitively Codable through some protocol but it must conform to `SerializationRequirement` to be “usable”…

Reply:
Yeah I see what you mean, it's probably not super useful unless the serialization requirement is also that... 

As a simple rule I guess I still somewhat like it though... we can think more about it though 💭